cmd/prometheus/main.go

```diff
@@ -359,6 +359,10 @@ func main() {
 		PreAction(agentOnlySetting()).
 		SetValue(&cfg.agent.MaxWALTime)
 
+	a.Flag("storage.agent.no-lockfile", "Do not create lockfile in data directory.").
+		PreAction(agentOnlySetting()).
```

Comment:
I don't find `agentOnlySetting` in main branch, am I missing something or does this PR need a rebase?

Reply:
Yep, needs a rebase. Working on that now.